diff -ur a/lib/cpp/CMakeLists.txt b/lib/cpp/CMakeLists.txt
--- a/lib/cpp/CMakeLists.txt	2019-08-02 08:07:04.000000000 +0200
+++ b/lib/cpp/CMakeLists.txt	2019-08-02 08:14:15.000000000 +0200
@@ -42,6 +43,7 @@
    src/thrift/protocol/TProtocol.cpp
    src/thrift/transport/TTransportException.cpp
    src/thrift/transport/TFDTransport.cpp
+   src/thrift/transport/TForkingServerSocket.cpp
    src/thrift/transport/TSimpleFileTransport.cpp
    src/thrift/transport/THttpTransport.cpp
    src/thrift/transport/THttpClient.cpp
@@ -53,6 +55,7 @@
    src/thrift/transport/TBufferTransports.cpp
    src/thrift/server/TConnectedClient.cpp
    src/thrift/server/TServerFramework.cpp
+   src/thrift/server/TSimpleForkableServer.cpp
    src/thrift/server/TSimpleServer.cpp
    src/thrift/server/TThreadPoolServer.cpp
    src/thrift/server/TThreadedServer.cpp
