diff -ur a/lib/cpp/src/thrift/stdcxx.h b/lib/cpp/src/thrift/stdcxx.h
--- a/lib/cpp/src/thrift/stdcxx.h	2019-12-17 11:44:18.000000000 +0100
+++ b/lib/cpp/src/thrift/stdcxx.h	2019-12-17 11:46:22.000000000 +0100
@@ -33,7 +33,9 @@
 #if (BOOST_VERSION <= 106500)
 #include <boost/tr1/functional.hpp>
 #else
-#include <tr1/functional>
+#include <boost/functional.hpp>
+#include <boost/bind.hpp>
+#include <boost/function.hpp>
 #endif
 #define _THRIFT_FUNCTIONAL_TR1_ 1
 #endif
@@ -42,19 +44,19 @@
 
   namespace apache { namespace thrift { namespace stdcxx {
 
-    using ::std::tr1::bind;
-    using ::std::tr1::function;
+    using boost::bind;
+    using boost::function;
 
     namespace placeholders {
-      using ::std::tr1::placeholders::_1;
-      using ::std::tr1::placeholders::_2;
-      using ::std::tr1::placeholders::_3;
-      using ::std::tr1::placeholders::_4;
-      using ::std::tr1::placeholders::_5;
-      using ::std::tr1::placeholders::_6;
-      using ::std::tr1::placeholders::_7;
-      using ::std::tr1::placeholders::_8;
-      using ::std::tr1::placeholders::_9;
+      using boost::placeholders::_1;
+      using boost::placeholders::_2;
+      using boost::placeholders::_3;
+      using boost::placeholders::_4;
+      using boost::placeholders::_5;
+      using boost::placeholders::_6;
+      using boost::placeholders::_7;
+      using boost::placeholders::_8;
+      using boost::placeholders::_9;
     } // apache::thrift::stdcxx::placeholders
   }}} // apache::thrift::stdcxx
 
